lighthouse-core/audits/webapp-install-banner.js

```diff
@@ -0,0 +1,110 @@
+ */
+
+'use strict';
+
+const Audit = require('./audit');
+const SWAudit = require('./service-worker');
+const Formatter = require('../report/formatter');
+
+/**
+ * @fileoverview
+ * Audits if a page is configured to prompt users with the webapp install banner.
+ * https://github.com/GoogleChrome/lighthouse/issues/23#issuecomment-270453303
+ *
+ * Requirements:
+ *   * manifest is not empty
+ *   * manifest has valid start url
+ *   * manifest has a valid name
+ *   * manifest has a valid shortname
+ *   * manifest display property is standalone, minimal-ui, or fullscreen
+ *   * manifest contains icon that's a png and size >= 192px
+ *   * SW is registered, and it owns this page and the manifest's start url
+ *   * Site engagement score of 2 or higher
+
+ * This audit covers these requirements with the following exceptions:
+ *   * it doesn't consider SW controlling the starturl
+ *   * it doesn't consider the site engagement score (naturally)
+ */
+
+class WebappInstallBanner extends Audit {
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'PWA',
+      name: 'webapp-install-banner',
+      description: 'User can be prompted to Install the Web App',
+      helpText: 'While users can manually add your site to their homescreen, the [prompt (aka app install banner)](https://developers.google.com/web/updates/2015/03/increasing-engagement-with-app-install-banners-in-chrome-for-android) will proactively prompt the user to install the app if the various requirements are met and the user has moderate engagement with your site.',
+      requiredArtifacts: ['URL', 'ServiceWorker', 'Manifest']
+    };
+  }
+
+  static assessManifest(manifestValues, failures) {
+    if (manifestValues.isParseFailure) {
+      failures.push(manifestValues.parseFailureReason);
+      return;
+    }
+
+    const bannerCheckIds = [
+      'hasName',
+      'hasShortName',
+      'hasStartUrl',
+      'hasPWADisplayValue',
+      'hasIconsAtLeast192px'
+    ];
+    manifestValues.allChecks
+      .filter(item => bannerCheckIds.includes(item.id))
+      .forEach(item => {
+        if (item.passing === false) {
+          failures.push(item.userText);
+        }
+      });
+  }
+
+
+  static assessServiceWorker(artifacts, failures) {
+    const hasServiceWorker = SWAudit.audit(artifacts).rawValue;
+    if (!hasServiceWorker) {
+      failures.push('Site registers a Service Worker');
+    }
+  }
+
+  static audit(artifacts) {
+    const failures = [];
+
+    return artifacts.requestManifestValues(artifacts.Manifest).then(manifestValues => {
+      // 1: validate manifest is in order
+      WebappInstallBanner.assessManifest(manifestValues, failures);
+      // 2: validate we have a SW
+      WebappInstallBanner.assessServiceWorker(artifacts, failures);
+
+      const extendedInfo = {
+        value: {failures, manifestValues},
+        formatter: Formatter.SUPPORTED_FORMATS.NULL
+      };
+
+      // If we fail, share the failures
```

Comment:
the last part of all these audits is the same. I wonder if there's a way to dry that up a bit (possibly other duplicated places). One idea is for these new audits to extend a base audit like Rob did for the a11y audits.